pkg/kube/deployments.go

```diff
@@ -45,17 +45,17 @@ func GetDeploymentOwnerReference(dep *appsv1.Deployment) metav1.OwnerReference {
 	}
 }
 
-func DeleteDeployment(name string, namespace string, cli kubernetes.Interface) error {
-	_, err := cli.AppsV1().Deployments(namespace).Get(name, metav1.GetOptions{})
+func DeleteDeployment(name string, cli types.Deployments) error {
+	dep, _, err := cli.GetDeployment(name, &metav1.GetOptions{})
 	if err == nil {
-		err = cli.AppsV1().Deployments(namespace).Delete(name, &metav1.DeleteOptions{})
+		err = cli.DeleteDeployment(dep, &metav1.DeleteOptions{})
 	}
 	return err
 }
 
 // TODO, pass full client object with namespace and clientset
-func GetDeployment(name string, namespace string, cli kubernetes.Interface) (*appsv1.Deployment, error) {
-	existing, err := cli.AppsV1().Deployments(namespace).Get(name, metav1.GetOptions{})
+func GetDeployment(name string, cli types.Deployments) (*appsv1.Deployment, error) {
```

Comment:
These changes - while they may be justified - seem to me to essentially invert the role of the kube package. It is no longer code tied to kubernetes, but potentially a more generic layer above the interface behind which an implementation using the kubernetes API server would sit. Is this intended? What do we see as the role going forward for the kube package?

Reply:
The way I see it is that the `kube` package is still responsible for providing mechanisms to manipulate
common kubernetes resources, but it delegates the CRUD operations to a concrete implementation which
might not necessarily perform that against a Kubernetes cluster.